python/lsst/daf/butler/_butler_collections.py

```diff
@@ -280,7 +283,7 @@ def query_info(
         include_parents: bool = False,
         include_summary: bool = False,
         include_doc: bool = False,
-        summary_datasets: Iterable[DatasetType] | None = None,
+        summary_datasets: Iterable[DatasetType] | Iterable[str] | None = None,
```

Comment:
Should we allow `Iterable[str]` only to avoid all complications? I think that clients of this method that actually pass `summary_datasets` should already have a list of names for other purposes. I guess this also means that datasets manager has to accept `Iterable[str]` instead of `Iterable[DatasetType]`?

Reply:
I was thinking the same thing, but the convention in the rest of the new query system is that dataset types can be specified as either the name or `DatasetType` instance so it makes sense to follow that.  It's also common for this parameter to come from the result of a queryDatasetTypes call.